### PR TITLE
Search books

### PR DESCRIPTION
**Pull request by DragisaDraskovic**
Console.error cu prepraviti i napravicu toast notifications, ali bi prvo napravio osnovne funkcionalnosti pa bi se onda vratio na ovaj problem.

### CODE REVIEW

File: src/components/bookList/BookList.tsx [hunk 1]
```diff
@@ -1,52 +1,69 @@
...
+  const [ books, setBooks ] = useState<Book[]>([])
   const [ hasMore, setHasMore ] = useState(true)
-  const [ pageNubmer, setPageNumber ] = useState(1)
-
-  useEffect(() => {
-    getDataBooks()
-  }, [ pageNubmer ])
+  const [ pageNumber, setPageNumber ] = useState(1)
+  const [ searchValue, setSearchValue ] = useState('')
+  const [ currentSearch, setCurrentSearch ] = useState<string>(searchValue)
```
Comment (sandra-matkovic-valcon) [inline src/components/bookList/BookList.tsx:20]: You don't have to explicitly set basic types, TS knows that

File: src/components/bookList/BookList.tsx [hunk 2]
```diff
@@ -1,52 +1,69 @@
...
-      PageNumber: pageNubmer,
-      PageLength: numberOfCard
-    }
-    try{
-      const response = await BookService.getAllBooks(bookRequest)
-      setHasMore(pageNubmer * numberOfCard <= response.data.TotalCount)
-      setBook([ ...book,...response.data.Items ])
-    } catch(error) {
-      console.error(error)
+  const dataBooks = () => {
```
Comment (sandra-matkovic-valcon) [inline src/components/bookList/BookList.tsx]: Not very intuitive name for this function, could you name it somehow to describe what is that function doing?

File: src/components/bookList/BookList.tsx (hunk 1 above)
Comment (sandra-matkovic-valcon) [inline src/components/bookList/BookList.tsx:20]: Do you really need both searchValue and currentSearch?

File: src/services/BookService.tsx [hunk 3]
```diff
@@ -1,22 +1,37 @@
...
+      result += `&where=${JSON.stringify(where)}`
+    }
+  })
+  return result
 }
-const getAllBooks = ( bookRequest : BookRequest) : Promise<AxiosResponse<BookResponse>> => {
+
+const getAllBooks = ( getBookRequest : GetBooksProps) : Promise<AxiosResponse<BookResponse>> => {
   const config = {
     headers: {
```
Comment (sandra-matkovic-valcon) [inline src/services/BookService.tsx]: You have this config in all services, can you make it somehow generic? Extract it to some file and import it when needed, add extra properties when needed?

**Review by sandra-matkovic-valcon — CHANGES_REQUESTED** [review]

File: src/components/bookList/BookList.tsx [hunk 4]
```diff
@@ -1,52 +1,69 @@
...
+      PageLength: initialItem,
+      Where: [ { Field: 'Title',Value: searchValue, Operation: 2 } ]
     }
+    BookService.getAllBooks(bookRequest)
+      .then((response:AxiosResponse<BookResponse>) => {
+        const totalElements = response.data.TotalCount
+        setHasMore(pageNumber * initialItem <= totalElements)
+        setBooks((books) => [ ...books,...response.data.Items ])
+      })
+      .catch((e) => console.error (e))
```
Comment (sandra-matkovic-valcon) [inline src/components/bookList/BookList.tsx:38]: Handle errors differently, console.error is not the best way, you dont give any feedback to user that something went wrong...

**Review by sandra-matkovic-valcon — CHANGES_REQUESTED** [review]

File: src/components/bookList/BookList.tsx (hunk 2 above)
Comment (DragisaDraskovic) [reply src/components/bookList/BookList.tsx]: Ok i fix this.

File: src/components/bookList/BookList.tsx (hunk 1 above)
Comment (DragisaDraskovic) [reply src/components/bookList/BookList.tsx:20]: Maybe not, but i just can implement my login with both state...

File: src/components/bookList/BookList.tsx (hunk 4 above)
Comment (DragisaDraskovic) [reply src/components/bookList/BookList.tsx:38]: Yes i know, I fix this problem when i finished return and rent features.

Comment (sandra-matkovic-valcon) [reply src/components/bookList/BookList.tsx:38]: But why dont you implement better solution for this new code and then later you have less places to change? You can change it later but isn't it more work than changing it as you work on these files?

File: src/services/BookService.tsx (hunk 3 above)
Comment (DragisaDraskovic) [reply src/services/BookService.tsx]: I fixed this, i make generic config file and i use it.

File: src/components/bookList/BookList.tsx (hunk 1 above)
Comment (DragisaDraskovic) [reply src/components/bookList/BookList.tsx:20]: Ok. I don't know this.

File: src/components/bookList/BookList.tsx [hunk 5]
```diff
@@ -1,52 +1,69 @@
...
-import BookRequest from '../../model/BookRequest'
 import BookService from '../../services/BookService'
 import Card from '../card/Card'
 import './BookList.css'
+import Search from '../search/Search'
+import Where from '../../model/Where'
+import BookResponse from '../../model/BookResponse'
 
-const numberOfCard = 24
+interface BooksRequest {
```
Comment (vladimir-agbaba-valcon) [inline src/components/bookList/BookList.tsx]: Why did you create this interface at the component if you already have it at BookRequest file at models folder?

File: src/components/bookList/BookList.tsx [hunk 6]
```diff
@@ -1,52 +1,69 @@
...
+  PageNumber: number,
+  PageLength: number,
+  Where: Where[],
+}
+const initialItem = 6
```
Comment (vladimir-agbaba-valcon) [inline src/components/bookList/BookList.tsx]: Initial item what? Not sure what is this const used for

File: src/components/search/Search.tsx [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+interface SearchProps {
+  setSearchValue: React.Dispatch<React.SetStateAction<string>>
+}
+const Search = ({ setSearchValue } : SearchProps) => {
+
+  const handleSeach = ({ target }: ChangeEvent<HTMLInputElement>) =>{
+    setSearchValue(target.value)
+  }
+  const debouncedChangeHandler = useMemo(
```
Comment (vladimir-agbaba-valcon) [inline src/components/search/Search.tsx:15]: Why use memo?

File: src/services/BookService.tsx [hunk 8]
```diff
@@ -1,31 +1,39 @@
 import axios, { AxiosResponse } from 'axios'
 
 import TokenService from './TokenService'
-import BookRequest from '../model/BookRequest'
 import BookResponse from '../model/BookResponse'
+import Where from '../model/Where'
+import config from '../utils/Config'
 
-const token = TokenService.getAccesToken()
+// const token = TokenService.getAccesToken()
```
Comment (vladimir-agbaba-valcon) [inline src/services/BookService.tsx]: Commented code

**Review by sandra-matkovic-valcon — APPROVED** [review]
Ok for my requests but first take a look at requests from @vladimiragbaba98 then merge

File: src/components/bookList/BookList.tsx (hunk 5 above)
Comment (DragisaDraskovic) [reply src/components/bookList/BookList.tsx]: Mistake, i fix this.

File: src/components/bookList/BookList.tsx (hunk 6 above)
Comment (DragisaDraskovic) [reply src/components/bookList/BookList.tsx]: This is initial number for render card to main page. I changed name for this const.

File: src/components/search/Search.tsx (hunk 7 above)
Comment (DragisaDraskovic) [reply src/components/search/Search.tsx:15]: I don't know, i found this code on internet and I implement it. This code working.

File: src/services/BookService.tsx (hunk 8 above)
Comment (DragisaDraskovic) [reply src/services/BookService.tsx]: I fix this.

**Review by vladimir-agbaba-valcon — APPROVED** [review]

**Pull request merged**